homeassistant/components/media_source/local_source.py

```diff
@@ -56,32 +37,81 @@ def __init__(self, hass: HomeAssistant):
     @callback
     def async_full_path(self, source_dir_id, location) -> Path:
         """Return full path."""
-        return self.hass.config.path("media", location)
+        return Path(self.hass.config.media_dirs[source_dir_id], location)
+
+    @callback
+    def async_parse_identifier(self, item: MediaSourceItem) -> Tuple[str, str]:
+        """Parse identifier."""
+        if not item.identifier:
+            source_dir_id = ""
+            location = ""
```

Comment:
We don't need to do the sanitize path check, as we know it's valid and so can return right away
```suggestion
            # Empty source_dir_id and location
            return "", ""
```

Reply:
Oh this will cause a lint issue with unnecessary `else` below.

Reply:
yeah adjusted 👍